### PR TITLE
Get the status of pd, tikv and tidb to coordinate these servers

### PR DESCRIPTION
**Pull request by PHILO-HE**
Remove libtidb.jar. So temporarily tidb doesn't support SSM in trunk branch.

### CODE REVIEW

File: smart-server/src/main/java/org/smartdata/server/SmartServer.java [hunk 1]
```diff
@@ -130,11 +130,13 @@ public static StartupOption processArgs(String[] args, SmartConf conf) throws Ex
...
   static SmartServer processWith(StartupOption startOption, SmartConf conf) throws Exception {
     if (isTidbEnabled(conf)) {
-      Thread db = new Thread(new Launch());
+      LaunchDB launchDB = new LaunchDB();
+      Thread db = new Thread(launchDB);
       LOG.info("Starting PD, TiKV and TiDB..");
       db.start();
       try {
-        Thread.sleep(12000);
+        while (!launchDB.isCompleted())
```
Comment (huafengw) [inline smart-server/src/main/java/org/smartdata/server/SmartServer.java]: Don't omit the braces even if there is only one line of code is while-loop

File: smart-tidb/pom.xml [hunk 2]
```diff
@@ -30,55 +30,55 @@
...
+                            <!--<arguments>-->
+                                <!--<argument>-r</argument>-->
+                                <!--<argument>${basedir}/src/main/resources/META-INF/</argument>-->
+                            <!--</arguments>-->
+                        <!--</configuration>-->
+                    <!--</execution>-->
+                <!--</executions>-->
+            <!--</plugin>-->
+        <!--</plugins>-->
+    <!--</build>-->
```
Comment (huafengw) [inline smart-tidb/pom.xml]: If these are not needed, just delete them all.

File: smart-tidb/src/main/java/org/smartdata/tidb/LaunchDB.java [hunk 3]
```diff
@@ -20,28 +20,49 @@
...
+  public boolean isCompleted() {
+    return dbReady;
+  }
 
+  public void run() {
     Thread pdThread = new Thread(pdServer);
     pdThread.start();
     try {
-      Thread.sleep(4000);
+      while (!pdServer.isReady())
```
Comment (huafengw) [inline smart-tidb/src/main/java/org/smartdata/tidb/LaunchDB.java]: Same problem

File: smart-engine/src/main/java/org/smartdata/server/engine/rule/RuleExecutor.java [hunk 4]
```diff
@@ -156,17 +156,22 @@ public String genVirtualAccessCountTable(List<Object> parameters) {
...
     List<String> tableNames =
         getAccessCountTablesDuringLast(interval);
-    return generateSQL(tableNames, newTable, countFilter);
+    return generateSQL(tableNames, newTable, countFilter, adapter);
   }
 
   @VisibleForTesting
-  static String  generateSQL(List<String> tableNames, String newTable, String countFilter) {
-    String sqlFinal;
+  static String  generateSQL(List<String> tableNames, String newTable, String countFilter, MetaStore adapter) {
```
Comment (huafengw) [inline smart-engine/src/main/java/org/smartdata/server/engine/rule/RuleExecutor.java]: The call stack of this function is really complicated, hope someone can refactor it in the future :(

**Comment by PHILO-HE** [comment]
Huafeng, I have modified the code according to your comments. Thanks!

**Comment by huafengw** [comment]
@littlezhou Please help to check the modified RuleExcutor code

**Pull request merged**